app/src/main/java/net/squanchy/service/firebase/FirebaseDbService.kt

```diff
@@ -79,8 +80,12 @@ class FirebaseDbService(private val database: DatabaseReference) {
                     if (emitter.isDisposed) {
                         return
                     }
-                    val value = dataSnapshot.getValue(clazz)
-                    emitter.onNext(map(value))
+                    try {
+                        val value = dataSnapshot.getValue(clazz)
+                        emitter.onNext(map(value))
+                    } catch (e: Exception) {
```

Comment:
Gotta catch 'em all!

Because really, for *anything* that might go wrong, we need more info than the awful errors that come out otherwise.

Reply:
Should we also emit the error on `onError`?